src/V3EmitCFunc.h

```diff
@@ -1427,6 +1437,35 @@ class EmitCFunc VL_NOT_FINAL : public EmitCConstInit {
         UASSERT_OBJ(!nodep->mTaskBodiesp(), nodep, "These should have been lowered");
         iterateChildrenConst(nodep);
     }
+    void visit(AstAssertCtl* nodep) override {
+        switch (nodep->ctlType()) {
+        case VAssertCtlType::OFF:
+        case VAssertCtlType::ON: {
+            if (nodep->hierarchicalNames().empty() || nodep->name().empty()) break;
+
+            const auto arraySize = std::to_string(nodep->hierarchicalNames().size());
+            string assertName = nodep->name() + std::to_string(m_assertCtlNum);
+            std::replace(assertName.begin(), assertName.end(), '.', '_');
+            const auto arrayName = assertName + "_assertions";
+
+            string stmt
+                = "constexpr std::array<const char*, " + arraySize + "> " + arrayName + " = {\n";
+            for (const auto& assertName : nodep->hierarchicalNames()) {
+                stmt += "\"" + assertName + "\",\n";
+            }
+            stmt += "};\n";
+            stmt += "for (const char* name : " + arrayName
+                    + ") {\n"
+                      "   vlSymsp->_vm_contextp__->assertOnFor(name, "
+                    + (nodep->ctlType() == VAssertCtlType::ON ? "true" : "false") + ");\n";
+            stmt += "}\n";
```

Comment:
Seems strange we emit a runtime loop.  I'd think we either can expand the loop at Verilation time, or we call a include/ function to do this rather than "inline" the loop.

Reply:
Are you thinking about emitting those calls as AstWhile/AstFor constructs or by unrolling them manually (emitting C++ code without for-loop)?

Reply:
I was thinking you either call one function in Verilated with the loop, or inline the expanded checks without the loop. Probably former is better for code space.